TAN-5715 User location on idea map

IvaKop opened the pull request:
# Changelog
## Changed
- The user location will now be visible on the ideas map view provided the user gives location permission to the browser

IvaKop committed:
Add user location to map

IvaKop committed:
Only show it on idea map

IvaKop committed:
Use watch position to ensure continous updates

IvaKop committed:
Simplify useEffect to handle user location

IvaKop commented:
@amanda-anderson A quick functional test would be appreciated for this one! Thanks 🙌

amanda-anderson reviewed (commented):
@IvaKop From my functional test:

I didn't get the "Location Requested" permission popup when I loaded the page. I went into my browser settings instead though and enabled it there so I could test, which did make the location icon on the map appear. Not sure why it didn't ask me for permission though.

However the location is really far away from where I actually am in the city :thinking: I'm wondering if this only really makes sense for Mobile devices, where we can actually get an accurate result? Otherwise I don't find this super useful (and instead a bit confusing) on Desktop.

Also, the circle icon is quite similar to the Idea marker (especially if a client is using a blue branding colour). It's indeed a bit smaller, but I think users might potentially confuse them?

<img width="1738" height="752" alt="image" src="https://github.com/user-attachments/assets/a5198044-df10-484b-9728-6761b3d07455" />

IvaKop commented:
@amanda-anderson That's great, thanks for letting me know. For me, the location is really accurate and the popup appears with no issues so it's really useful to know it's not working so well for you. We will discuss it within the team and see where we go from there.

IvaKop committed:
Show location only on small screens and change idea shape to make it different from location

IvaKop committed:
Only change idea shape for smaller screens

IvaKop commented:
@amanda-anderson I implemented some changes based on your comments:
- The location is now only available on tablet or smaller. Debated a bit whether to add actual device detection instead of basing this on the screen size but decided this is probably good enough for now and we can always change it, if we see it's creating issues
- I changed the shape of the ideas on the map to triangle on smaller devices so that it doesn't look super similar to the location dot

I also tested this on my actual phone and for me it worked alright. Could you maybe do the same and see if you still have location issues. I made an [epic deployment](https://location.epic.govocal.com/en/) for this.

amanda-anderson commented:
@IvaKop Yes works much better on my mobile (gets the right location) :rocket: 

One thing I noticed is that when I selected an idea (triangle), the location circle disappeared:
<img width="300"  alt="IMG_2512" src="https://github.com/user-attachments/assets/d72b308e-ff59-4d44-b4e8-39a942e42f64" />
<img width="300"  alt="IMG_2511" src="https://github.com/user-attachments/assets/ef10d1f4-e7cb-4dd4-8003-9fe18140a86c" />

amanda-anderson reviewed (approved):
LGTM! Left one comment from the functional review, but happy to approve in advance :)

IvaKop committed:
Merge branch 'master' into TAN-5715-user-location

IvaKop committed:
Fix disappearing location point

IvaKop commented:
@amanda-anderson Nice catch! Fixed it :)
